### PR TITLE
Add new GameServer lifecycle diagrams for Reserved

### PR DESCRIPTION
**Pull request by markmandel**
Shows how the workflow can exist for 1st party/distributed matchmakers.

Work on #660

### CODE REVIEW

File: build/Makefile [hunk 1]
```diff
@@ -392,8 +392,6 @@ gen-crd-client: $(ensure-build-image)
 	docker run --rm $(common_mounts) -w $(workdir_path) $(build_tag) /root/gen-crd-client.sh
 	docker run --rm $(common_mounts) -w $(workdir_path)/pkg $(build_tag) goimports -w .
 
-site-images: $(site_path)/static/diagrams/gameserver-states.dot.png $(site_path)/static/diagrams/gameserver-lifecycle.puml.png
-
 # generate pngs from dot files
```
Comment (roberthbailey) [inline build/Makefile]: Should this part of the makefile move as well? Are there places outside of the website where we generate pngs from dot files and puml files?

File: site/static/diagrams/gameserver-reserved.puml [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+  available for, so it will not be
+  scaled down
+end note
+Binary -> Matchmaker: Register()
+note right
+  Register with the Matchmaker as
+  available to host a game session
+end note
+note left
+match maker registration
```
Comment (roberthbailey) [inline site/static/diagrams/gameserver-reserved.puml]: Start with a capital to be consistent with the other notes.

Comment (roberthbailey) [inline site/static/diagrams/gameserver-reserved.puml]: Is match maker one work (e.g. matchmaker)?

File: site/static/diagrams/gameserver-reserved.puml [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+  the configured threshold.
+end note
+GameServer <-- SDK: Maintains Healthy status
+
+== Match Maker Registration ==
+group loop
+Binary -> SDK: SDK.Reserve(duration)
+GameServer <-- SDK: Set state to <i>Reserved</i> for duration
+note right
+  <i>duration</i> should be longer
```
Comment (roberthbailey) [inline site/static/diagrams/gameserver-reserved.puml]: Start with a capital to be consistent with the other notes.

File: build/Makefile (hunk 1 above)
Comment (markmandel) [reply build/Makefile]: That is true that we don't - but it seemed general enough that maybe leave it where it was.

Honestly, I've no strong opinion on this, so if you feel strongly, I'll move it.

File: site/static/diagrams/gameserver-reserved.puml (hunk 2 above)
Comment (markmandel) [reply site/static/diagrams/gameserver-reserved.puml]: Yep, good catch.

**Comment by markmandel** [comment]
Changes made, per comments :+1:

File: build/Makefile (hunk 1 above)
Comment (roberthbailey) [reply build/Makefile]: If we are still thinking about moving the website to a different repo after 1.0 I'd move it to the site makefile now to make that simpler.

Comment (markmandel) [reply build/Makefile]: Makes perfect sense. Will make the move :+1:

**Pull request merged**